### PR TITLE
Add component status and owners information for individual projects

### PR DESCRIPTION
**Pull request by vishweshbankwar**
Fixes #
Design discussion issue #

## Changes

* Added project status section in the main Readme for clarifying project status and definitions for alpha, beta, rc and stable versions.
* Updates individual readme to specify the status and individual owners (currently it is not easy to look up the owners).

if the direction looks good, will update the rest of the projects as well. This is manual right now but can be automated later.

Borrowed the idea and some content from https://github.com/open-telemetry/opentelemetry-collector-contrib

## Merge requirement checklist

* [ ] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/main/CONTRIBUTING.md) guidelines followed (license requirements, nullable enabled, static analysis, etc.)
* [ ] Unit tests added/updated
* [ ] Appropriate `CHANGELOG.md` files updated for non-trivial changes
* [ ] Changes in public API reviewed (if applicable)

### CODE REVIEW

File: README.md
Comment (CodeBlanch) [inline README.md:18]: @vishweshbankwar I went through and did a review of the new content. Here are the tweaks I made. Feel free to take or ignore what you want 😄

```markdown
## Project status

This repository is a collection of unrelated components maintained by different authors and groups. As such components shipped from this repository (via [Nuget](https://www.nuget.org/))may be at different stability/maturity levels. The
status for each individual component SHOULD be mentioned in its respective `README.md` file and SHOULD fall into one of the following categories:

### Development

Component is currently in development and is NOT available on
[Nuget](https://www.nuget.org/).

### Alpha

The component is ready to be used for limited non-critical workloads and the
authors of this component would welcome your feedback. Bugs and performance
problems should be reported, but component owners might not work on them right
away. Components can go through significant breaking changes and there are no
backward compatibility guarantees. Package versions in this status have the
`-alpha` extension.

### Beta

Same as Alpha, but comparatively more stable. Package versions in this status have the
`-beta` extension.

### Release candidate

Component is close to stability. There may be minimal breaking changes between
releases. A component at this stage is expected to have had exposure to
non-critical production workloads already during its **Alpha/Beta** phase(s),
making it suitable for broader usage. Package versions in this status have the
`-rc` extension.

### Stable

The component is ready for general availability. Bugs and performance problems
should be reported and the component owner(s) SHOULD triage and/or resolve them in a timely manner. Breaking changes, including configuration options and the component's output (semantic conventions, etc.), are only allowed across major version bumps unless under special circumstances such as security related fixes.
```

File: src/OpenTelemetry.Resources.Azure/README.md
Comment (cijothomas) [inline src/OpenTelemetry.Resources.Azure/README.md:6]: One more thing to note here:
If the code owners are OTel .NET maintainer and/or approver, have a way to indicate that. End users looking at this repo would usually want to know if the cmponent is maintained by repo maintainres/approvers or other individuals.

File: README.md
Comment (cijothomas) [inline README.md:55]: simply state, the component is expected to follow semver20

File: src/OpenTelemetry.Resources.Azure/README.md
Comment (vishweshbankwar) [reply src/OpenTelemetry.Resources.Azure/README.md:6]: for those cases I was just planning to put `@open-telemetry/dotnet-contrib-maintainers` for code owners.

File: README.md
Comment (vishweshbankwar) [reply README.md:18]: Thanks @CodeBlanch!. Updated!

File: README.md
Comment (vishweshbankwar) [reply README.md]: updated

File: src/OpenTelemetry.Resources.Azure/README.md
Comment (cijothomas) [reply src/OpenTelemetry.Resources.Azure/README.md:6]: I think it is better to be very explicit about the relation with Otel owned groups.

OTel-DotNet maintainers, approvers
OTel-Dotnet-Contrib maintainers, approvers
OTel-Dotnet-Instrumentation - maintainers, approvers.

eg:
[@vishweshbankwar](https://github.com/vishweshbankwar) OTel-Dotnet Approver, OTel-Dotnet-Contrib Maintainer. 

Does this sound better or this is too much details?

File: README.md
Comment (cijothomas) [inline README.md:21]:
```suggestion
authors and groups. As such, components shipped from this repository (via
```

File: README.md
Comment (cijothomas) [inline README.md:23]:
```suggestion
The status for each individual component is mentioned in its respective
```

File: README.md
Comment (cijothomas) [inline README.md:24]:
```suggestion
`README.md` file and will fall into one of the following categories:
```

File: README.md
Comment (cijothomas) [inline README.md:38]:
```suggestion
`-alpha` extension (eg: opentelemetry.exporter.abc-1.0.0-alpha.1).
```

File: README.md
Comment (cijothomas) [inline README.md:60]: Line 67 is pointing to component_owenrs.yml. Can you update taht to say owners can be found from individual readme files?

File: src/OpenTelemetry.Resources.Azure/README.md
Comment (CodeBlanch) [reply src/OpenTelemetry.Resources.Azure/README.md:6]: Kind of a tricky little box here to fill 🤣 IMO what user cares most about: Is someone actively maintaining the thing? If yes, is it an official thing? Like if it says "AWS" does "AWS" actually own it/support it or was it contributed by some random person? Not sure how best to convey this info 🤔

How about a convention like...

```markdown
| Code Owners | Microsoft-supported (@vishweshbankwar, @rajkumar-rangaraj) |
| Code Owners | CompanyName-supported (@some_user_name)) |
| Code Owners | Community-supported (@some_otel_community_group1, @some_otel_community_group2)) |
| Code Owners | Individual-supported (@some_user_name) |
```

Comment (vishweshbankwar) [reply src/OpenTelemetry.Resources.Azure/README.md:6]: I think we should leave it at `@open-telemetry/dotnet-contrib-maintainers`. 

I don't think we can put company name here as that's driven by the support policies outside of OTel.

Comment (cijothomas) [reply src/OpenTelemetry.Resources.Azure/README.md:6]: > I think we should leave it at @open-telemetry/dotnet-contrib-maintainers. it's easy to see who the maintainers are.

It is easy, but not every component is owned by contrib maintainers right!

For example, https://github.com/open-telemetry/opentelemetry-dotnet-contrib/tree/main/src/OpenTelemetry.Exporter.Geneva is maintained by a set of people including Reiley. And Reiley is not part of contrib approvers/maintainers.

Lets not block on this for now, and start with something simpler, and later evolve to better convey "who maintains this component".

Comment (CodeBlanch) [reply src/OpenTelemetry.Resources.Azure/README.md:6]: > I don't think we can put company name here as that's driven by the support policies outside of OTel.

Not sure what that means TBH. Why couldn't we put Microsoft on something like GenevaExporter or OneCollectorExporter?

Expanding on what I said above, I don't think list of owners is really _that_ useful to anyone.  Sort of nice, but users probably just want to know one thing: What is the support around this component? That's really what we _should_ have in these READMEs perhaps.

Something like this would be awesome...

| Status        |           |
| ------------- |-----------|
| Stability     |  [Beta](..\..\Readme.md#beta)|
| Support policy | ...???... |
| Code Owners   |  ...list here... |

Consider three examples...

Component A:

| Status        |           |
| ------------- |-----------|
| Stability     |  [Stable](..\..\Readme.md#stable)|
| Support policy | [Supported for production](#support-policy) |
| Code Owners   |  Microsoft (list-of-owners) |

Component B:

| Status        |           |
| ------------- |-----------|
| Stability     |  [Stable](..\..\Readme.md#stable)|
| Support policy | [Limited support](#support-policy) |
| Code Owners   |  list-of-owners |

Component C:

| Status        |           |
| ------------- |-----------|
| Stability     |  [Stable](..\..\Readme.md#stable)|
| Support policy | n/a |
| Code Owners   |  @some-guy |

As a user I would find this type of info really helpful in making decisions about what to use. Doesn't have to be solved on this PR... just providing thoughts!

Comment (vishweshbankwar) [reply src/OpenTelemetry.Resources.Azure/README.md:6]: > For example, https://github.com/open-telemetry/opentelemetry-dotnet-contrib/tree/main/src/OpenTelemetry.Exporter.Geneva is maintained by a set of people including Reiley. And Reiley is not part of contrib approvers/maintainers.

I see - for such cases I was going to simply add the owners listed here: https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/main/.github/component_owners.yml

Planning to use `@open-telemetry/dotnet-contrib-maintainers` only for components which does not have owners listed atm e.g. https://github.com/open-telemetry/opentelemetry-dotnet-contrib/tree/main/src/OpenTelemetry.Instrumentation.AspNetCore.

Comment (rajkumar-rangaraj) [reply src/OpenTelemetry.Resources.Azure/README.md:6]: We can keep this simple, as done in the [collector contrib](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/exporter/azuremonitorexporter#azure-monitor-exporter), and enhance it later on a need basis.

I like this format proposed by Cijo - [@vishweshbankwar](https://github.com/vishweshbankwar) OTel-Dotnet Approver, OTel-Dotnet-Contrib Maintainer

Comment (vishweshbankwar) [reply src/OpenTelemetry.Resources.Azure/README.md:6]: had an offline discussion, keeping it `@open-telemetry/dotnet-contrib-maintainers` for now. Will need to revisit the points raised by @CodeBlanch on the support part.

**Comment by some-guy** [comment]
Hello folks!

I'm not sure why I'm on this list, I have no affiliation with opentelemetry!

File: README.md
Comment (cijothomas) [inline README.md:20]:
```suggestion
This repository is a collection of components maintained by different
```

File: src/OpenTelemetry.Extensions.Enrichment/README.md
Comment (cijothomas) [inline src/OpenTelemetry.Extensions.Enrichment/README.md:6]: lets do a follow up to see if there are owners, else start removing components.

Comment (vishweshbankwar) [reply src/OpenTelemetry.Extensions.Enrichment/README.md:6]: Yes!

File: src/OpenTelemetry.Extensions/README.md
Comment (cijothomas) [inline src/OpenTelemetry.Extensions/README.md:6]: https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/main/.github/component_owners.yml#L21-L23
Missing Mike.

File: src/OpenTelemetry.Instrumentation.AWSLambda/README.md
Comment (cijothomas) [inline src/OpenTelemetry.Instrumentation.AWSLambda/README.md:6]: https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/main/.github/component_owners.yml#L34 Christian missing

**Review by cijothomas — APPROVED** [review]
Thanks. this looks very good to start with. We can iterate based on feedbacks.
(Also, please double check if all owners are correctly listed by cross checking with the component owners file before merging).

**Pull request merged**
